src/app-config/posthog.ts

```diff
@@ -1,25 +1,56 @@
+import fs from 'fs-extra';
 import { PostHog, PostHogOptions } from 'posthog-node';
+import { v4 as uuidv4 } from 'uuid';
 
-type PostHogCliOptions = PostHogOptions & {
-  analyticsId: string;
-};
+type PostHogCliOptions = Omit<PostHogOptions, 'persistence'> & ({
+  persistence: 'memory'
+} | {
+  persistence: 'file';
+  propertiesFile: string;
+} | Record<string, never>);
 
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-ignore
 export class PostHogCli extends PostHog {
-  constructor(apiKey: string, private options: PostHogCliOptions) {
-    super(apiKey, options);
+  private _properties: Record<string, any | undefined>;
+
+  constructor(apiKey: string, private _options: PostHogCliOptions) {
+    super(apiKey, {
+      ..._options,
+      persistence: 'memory',
+    });
+
+    this._properties = {};
+    if (this._options.persistence === 'file' && fs.existsSync(this._options.propertiesFile)) {
+      this._properties = fs.readJSONSync(this._options.propertiesFile);
+    } else {
+      this.setPersistedProperty('anonymous_id', uuidv4());
+    }
   }
 
-  capture(message: { event: string; properties?: Record<string | number, any>; }): void {
-    // We override reInit so the distinctId isn't used
-    return super.capture({ distinctId: '<unused>', ...message });
+  /**
+   * @override
+   */
+  getPersistedProperty(key: string): any | undefined {
+    if (this._options.persistence === 'file') {
+      return this._properties[key];
+    } else {
+      return super.getPersistedProperty(key as any);
+    }
   }
 
-  // Override reInit to not set distinct_id every call
-  private reInit(distinctId: string): void {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore Set the anonymous_id so we are consistent across cli calls
-    this._sharedClient.setPersistedProperty('anonymous_id', this.options.analyticsId);
+  /**
+   * @override
+   */
+  setPersistedProperty(key: string, value: any | null): void {
+    super.setPersistedProperty(key as any, value);
+    this._properties[key] = value;
+
+    if (this._options.persistence === 'file') {
+      fs.ensureFileSync(this._options.propertiesFile);
+      fs.writeJSONSync(this._options.propertiesFile, this._properties);
+    }
+  }
+
+  capture(message: { event: string; properties?: Record<string | number, any>; }): void {
+    return super.capture({ distinctId: this.getPersistedProperty('anonymous_id'), ...message });
```

Comment:
Setting the distinctId to the anon will create a new user each time and won't allow for aliasing multiple times

Reply:
I think we might just need to override this method and use anon instead of distinct when necessary:
https://github.com/PostHog/posthog-js-lite/blob/cb050098714676b00ddbdae3a5a4e1bdf1feec11/posthog-core/src/index.ts#L145

Reply:
This would work for a single user's machine, but will create spam in recurring CI pipelines which is why I had to override in the first place.

Reply:
> This would work for a single user's machine, but will create spam in recurring CI pipelines which is why I had to override in the first place.

Totally forgot about this use case, but should we instead just force disable for CI? This flow works correctly for individual developer usage whereas the other one didn't.

Reply:
I'd be down to force disable for CI. Prob not the analytics you're looking for. We have the `is-ci` pkg

Reply:
It's not an ideal answer, but we already collect server side events so I don't think the CI events are worth prioritizing over individual user experiences. I'll look into it now.

Reply:
Although my pref would still be to not set distinctId to an anon value. It should be easier now since we have direct access to override the captureStateless and other methods. 

```ts
  protected captureStateless(
    distinctId: string,
    event: string,
    properties?: { [key: string]: any },
    options?: PosthogCaptureOptions
  ): this {
    const distinct_id = this.getPersistedProperty('distinct_id');
    if (distinct_id) {
      const payload = this.buildPayload({ distinct_id: distinctId, event, properties })
      this.enqueue('capture', payload, options)
    } else {
      const payload = this.buildPayload({ anonymous_id: anonymous_id, event, properties })
      this.enqueue('capture', payload, options)
    }

    return this
  }
  ```

Reply:
Using the anon-id is the exact way the browser-side versions of posthog work. The docs and marketing site don't have access to the user distinct ID, but the correlation still happens because we ran an "identify" in our web app. I think this is still the advertised way of doing it.

Reply:
> I'd be down to force disable for CI. Prob not the analytics you're looking for. We have the is-ci pkg

Added now

Reply:
I'm not sure why posthog-node doesn't inherit from this class, but this is what the web and react packages extend so there's easy `getDistinctId()` methods which falls back to the anon ID if there's nothing else there:

https://github.com/PostHog/posthog-js-lite/blob/master/posthog-core/src/index.ts#L606

Reply:
The docs and marketing sites don't use distinct id at all right? So we shouldn't be using it here when its not the distinct id. What do you think of my proposal for captureStateless?

Reply:
Yeah so we def want to use the anon id. It seems like an easy change and then we don't need to worry about disabling CI. Although up to you if you still want to disable. We could add a tag for isCI so we could filter out that usage.

Reply:
@mueschm looping you in here per @tjhiggins request. He has to take off and said that you can take over reviewing and merging as you see fit.

We spoke offline as well and concluded that the implementation in this PR seems correct. See this docs article which took us a while to find that confirms how the aliasing works:
https://posthog.com/docs/integrate/user-properties#how-are-properties-managed-when-merging-or-aliasing-users